### ISSUE TITLE
Problem with indent and wrapped lines

### ISSUE THREAD

**Issue by SteveGZr** [eslint/eslint#3603]
Hey there,

I got a problem with the indent rule and wrapped lines. 
I'm using Coding Conventions, where you wrap your lines when they're longer than 80 characters and when you wrap your lines you have to indent it by 4 characters. And the next line should be indented by 2 characters like every other.

Example:

````javascript
sample_for_js_tests(
    'description and name of this sample, which can be very long sometimes', {
  default_variable_1: 'foo',
  default_variable_2: 'bar',
});
````

This would be right, but the indent rule gives me this:

    error  Expected indentation of 6 space characters but found 2

Did I find an issue here, or does it just work as intended?

    > eslint -v
    v1.3.1

I am using

    "rules": {
        "indent": [2, 2]

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by BYK** [comment]
This looks like the expected behavior to me. @gyandeeps, thoughts?

**Comment by SteveGZr** [comment]
After choosing to not use double indentation on wrapped lines, I got another problem now.

I got code like this:

    it('foo should allow bar', function() {                                
      return expect(response).to.have.header(                                   
          'Example-Header', 'Content-Type, If-Match');            
    }); 

And ESLint is not showing me any error, I set indent on two, but there are four spaces..

**Comment by BYK** [comment]
@SteveGZr - I believe the second issue you reported is the same as #3591 which should be resolved soon.

**Comment by gyandeeps** [comment]
On the first issue, I want to know the line number before I can say anything here.

**Comment by SteveGZr** [comment]
1 sample_for_js_tests(
2     'description and name of this sample, which can be very long sometimes', {
3   default_variable_1: 'foo',
4   default_variable_2: 'bar',
5 });

third and fourth line

**Comment by gyandeeps** [comment]
As of right now, we have not implemented the multi-line indentation logic, with that being said the rules is behaving as expected because the object start `{` has a 4 level indentation so the body has to be 6.

**Comment by SteveGZr** [comment]
Alright, thanks for the answer.

**Comment by FND** [comment]
Are there any plans for supporting double indentation for line continuations?

**Comment by opike** [comment]
+1 on adding this option. google style guide advocates for double indentation when line wrapping: https://google.github.io/styleguide/jsguide.html#formatting-indent

**Comment by FND** [comment]
FWIW, v4.0's new indent rule supports double indentation in some places: https://github.com/FND/eslint-config/commit/58996c1b48539d4d10f82c1ec66a9a7cc9722282

However, due to this level of granularity, we end up with a piecemeal approach that still leaves various other scenarios where double indentation is not only not enforced, but indeed flagged as an error:

```javascript
let foo = condition ?
⋯   ⋯   bar(…) :
⋯   ⋯   baz(…);
```
